lib/block_view_lockups.go

```diff
@@ -0,0 +1,2178 @@
+	data = append(data, VariableEncodeUint256(txnData.LockupAmountBaseUnits)...)
+	return data, nil
+}
+
+func (txnData *CoinLockupMetadata) FromBytes(data []byte) error {
+	rr := bytes.NewReader(data)
+
+	// ProfilePublicKey
+	profilePublicKeyBytes, err := DecodeByteArray(rr)
+	if err != nil {
+		return errors.Wrapf(err, "CoinLockupMetadata.FromBytes: Problem reading ProfilePublicKey")
+	}
+	txnData.ProfilePublicKey = NewPublicKey(profilePublicKeyBytes)
+
+	// UnlockTimestampNanoSecs
+	uint64UnlockTimestampNanoSecs, err := ReadUvarint(rr)
+	if err != nil {
+		return errors.Wrapf(err, "CoinLockupMetadata.FromBytes: Problem reading UnlockTimestampNanoSecs")
+	}
+	txnData.UnlockTimestampNanoSecs = int64(uint64UnlockTimestampNanoSecs)
+
+	// LockupAmountBaseUnits
+	txnData.LockupAmountBaseUnits, err = VariableDecodeUint256(rr)
+	if err != nil {
+		return errors.Wrapf(err, "CoinLockupMetadata.FromBytes: Problem reading LockupAmountBaseUnits")
+	}
+
+	return nil
+}
+
+func (txnData *CoinLockupMetadata) New() DeSoTxnMetadata {
+	return &CoinLockupMetadata{}
+}
+
+//
+// TYPES: UpdateDAOCoinLockupParamsMetadata
+//
+
+type UpdateCoinLockupParamsMetadata struct {
+	// LockupYieldDurationNanoSecs and LockupYieldAPYBasisPoints describe a coordinate pair
+	// of (duration, APY yield) on a yield curve.
+	//
+	// A yield curve consists of a series of (duration, APY yield) points. For example,
+	// the following points describe a simple yield curve:
+	//              {(6mo, 3%), (12mo, 3.5%), (18mo, 4%), (24mo, 4.5%)}
+	//
+	// Assuming RemoveYieldCurvePoint is false:
+	//    The point (LockupYieldDurationNanoSecs, LockupYieldAPYBasisPoints)
+	//    is added to the profile's yield curve. If a point with the same duration already exists
+	//    on the profile's yield curve, it will be updated with the new yield.
+	//    Note if LockupYieldDurationNanoSecs=0, nothing is modified or added at t=0.
+	// Assuming RemoveYieldCurvePoint is true:
+	//    The point (LockupYieldDurationNanoSecs, XXX) is removed from the profile's yield curve.
+	//    Note that LockupYieldAPYBasisPoints is ignored in this transaction.
+	//
+	// By setting LockupYieldDurationNanoSecs to zero, the yield curve attached to the profile
+	// is left unmodified. In any UpdateDAOCoinLockupParams transaction looking to modify only
+	// LockupTransferRestrictions, LockupYieldDurationNanoSecs would be set to zero.
+	LockupYieldDurationNanoSecs int64
+	LockupYieldAPYBasisPoints   uint64
+	RemoveYieldCurvePoint       bool
+
+	// When NewLockupTransferRestrictions is set true, the TransferRestrictionStatus specified
+	// in the transaction is updated in the transactor's profile for locked coins.
+	// Any subsequent transfers utilizing the transactor's locked coins are validated against
+	// the updated locked transfer restriction status.
+	NewLockupTransferRestrictions   bool
+	LockupTransferRestrictionStatus TransferRestrictionStatus
```

Comment:
So just so I understand: When it's Unrestricted, that means I can technically transfer locked balances to other users. Is that roughly correct?